sound/soc/sof/intel/hda-loader.c

```diff
@@ -338,6 +338,11 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 		if (ret < 0) {
 			dev_err(sdev->dev, "error: iteration %d of load fw failed err: %d\n",
 				i, ret);
+
+			/* power down the cores before retrying again */
+			hda_dsp_core_reset_power_down(sdev,
+						      HDA_DSP_CORE_MASK(0) |
+						      HDA_DSP_CORE_MASK(1));
```

Comment:
what if there are 4 cores? This doesn't scale to newer platforms.

Reply:
Also not convinced by the flow: you have two "continue" statements in this loop but only power down for the second case. This doesn't seem balanced or bullet-proof.

Reply:
> what if there are 4 cores? This doesn't scale to newer platforms.

Yes, I'll fix this.

Reply:
> Also not convinced by the flow: you have two "continue" statements in this loop but only power down for the second case. This doesn't seem balanced or bullet-proof.

@plbossart Oh the first step in theloopcl_dsp_init, powers down the cores in case of an error. Thats why I didnt do it there.